src/bio/bio_config.c

```diff
@@ -936,11 +956,68 @@ bio_read_rpc_srv_settings(const char *nvme_conf, bool *enable, const char **sock
 	*enable = rpc_srv_settings.enable;
 	*sock_addr = rpc_srv_settings.sock_addr;
 
-	D_DEBUG(DB_MGMT, "'%s' read from config: enabled=%d, addr %s\n",
-		NVME_CONF_SET_SPDK_RPC_SERVER, *enable, (char *)*sock_addr);
+	D_INFO("'%s' read from config: enabled=%d, addr %s\n", NVME_CONF_SET_SPDK_RPC_SERVER,
+	       *enable, (char *)*sock_addr);
 out:
 	if (cfg.method != NULL)
 		D_FREE(cfg.method);
+	/* Decode functions return positive RC for success or not-found */
+	if (rc > 0)
+		rc = 0;
+	return rc;
+}
+
+/**
+ * Set output parameters based on JSON config settings for NVMe auto-faulty feature and threshold
+ * criteria.
+ *
+ * \param[in]	nvme_conf	JSON config file path
+ * \param[out]	enable		Flag to enable the auto-faulty feature
+ * \param[out]	max_io_errs	Max IO errors (threshold) before marking as faulty
+ * \param[out]	max_csum_errs	Max checksum errors (threshold) before marking as faulty
+ *
+ * \returns	 Zero on success, negative on failure (DER)
+ */
+int
+bio_read_auto_faulty_criteria(const char *nvme_conf, bool *enable, uint32_t *max_io_errs,
+			      uint32_t *max_csum_errs)
+{
+	struct config_entry     cfg                  = {};
+	struct auto_faulty_info auto_faulty_criteria = {};
+	int                     rc;
+
+	rc = decode_daos_data(nvme_conf, NVME_CONF_SET_AUTO_FAULTY, &cfg);
+	if (rc != 0)
+		goto out;
+
+	rc = spdk_json_decode_object(cfg.params, auto_faulty_decoders,
+				     SPDK_COUNTOF(auto_faulty_decoders), &auto_faulty_criteria);
+	if (rc < 0) {
+		D_ERROR("Failed to decode '%s' entry: %s)\n", NVME_CONF_SET_AUTO_FAULTY,
+			spdk_strerror(-rc));
+		D_GOTO(out, rc = -DER_INVAL);
+	}
+
+	*enable = auto_faulty_criteria.enable;
+	if (*enable == false) {
+		*max_io_errs   = UINT32_MAX;
+		*max_csum_errs = UINT32_MAX;
+		goto out;
+	}
+	*max_io_errs = auto_faulty_criteria.max_io_errs;
+	if (*max_io_errs == 0)
+		*max_io_errs = UINT32_MAX;
+	*max_csum_errs = auto_faulty_criteria.max_csum_errs;
+	if (*max_csum_errs == 0)
+		*max_csum_errs = UINT32_MAX;
+
+out:
+	D_INFO("NVMe auto faulty is %s. Criteria: max_io_errs:%u, max_csum_errs:%u\n",
+	       *enable ? "enabled" : "disabled", *max_io_errs, *max_csum_errs);
+
+	if (cfg.method != NULL)
```

Comment:
Nit, not needed to test as D_FREE will do it.

Reply:
will fix if repushed